bandit.py

```diff
@@ -0,0 +1,45 @@
+#! /usr/bin/env python3
+
+import numpy as np
+import pandas as pd
+from scipy.special import softmax
+
+
+class Bandit(object):
+    def __init__(self, n_features, n_arms, step_size=0.05, baseline=False):
+        self.n_features = n_features
+        self.n_arms = n_arms
+        self.step_size = step_size
+        self.baseline = baseline
+        self.reward_sum = 0
+        self.iters = 0
+
+        # start at uniform
+        self.theta = np.zeros((self.n_features, self.n_arms))
+        
+    @property
+    def current_baseline(self):
+        return (self.reward_sum / self.iters
+                if self.baseline and self.iters
+                else 0)
+        
+    def predict_proba(self, X):
+        return softmax((X @ self.theta).reshape(-1, self.n_arms), axis=1)
+    
+    def predict(self, X, deterministic=True):
+        predictions = self.predict_proba(X)
+        if deterministic:
+            return predictions.argmax()
```

Comment:
this should be predictions.argmax(axis=1) so you get a vector

Reply:
Nice catch, thanks!